File: cpp/1652.cpp
#include <iostream>
#include <vector>
#include <numeric>
using namespace std;

class Solution {
public:
    vector<int> decrypt(vector<int>& code, int k) {
        int len = code.size();
        vector<int> ans(len);
        int r = k > 0 ? k + 1 : len;
        k = abs(k);
        int s = accumulate(code.begin() + r - k, code.begin() + r, 0);
        for (int i = 0; i < len; ++i) {
            ans[i] = s;
            s += code[r % len] - code[(r - k) % len];
            ++r;
        }
        return ans;
    }
};

File: python3/1652.py
from typing import List
class Solution:
    def decrypt(self, code: List[int], k: int) -> List[int]:
        n = len(code)
        ans = [0] * n
        r = k + 1 if k > 0 else n
        k = abs(k)
        s = sum(code[r - k : r])
        for i in range(n):
            ans[i] = s
            s += code[r % n] - code[(r - k) %n]
            r += 1
        return ans